docs/multi-tenancy.md

```diff
@@ -0,0 +1,129 @@
+# Multi-tenancy overview
+
+Logging operator supports sevaral multi-tenant scenarios
+
+## Multi tenancy behaviour and basic primitives
+
+### Structural components
+
+All container logs are aggregated into fluentd/syslogng (statefulset) by fluentbit (daemonset) without filtering or routing.
+
+### Logical components
+
+> Note: the notion of _flow_, _clusterflow_, _output_ and _clusteroutput_ are used interchangeably for fluentd and syslog-ng flows and outputs as well.
+
+- Logs can only be manipulated through higher level abstractions, called _flows_ and _outputs_ which are applied at the aggregator level.
+- `Flow` and `Output` are namespaced resources and can only work with logs in their own namespace, which is enforced in the aggregator.
+- `ClusterFlow` can work with logs from multiple namespaces as long as the logs are available in the aggregator.
+- `ClusterOutputs` can be used from a _clusterflow_ or from a namespaced _flow_.
+- All the above resources are combined in a single aggregator defined by the `Logging` resource.
+
+### Isolation options with **logging ref** and **watch namespaces**
+
+The `loggingRef` field in each resource makes it possible to create multiple logging "domains" with different configuration. 
+
+By default, resources with an empty `loggingRef` are processed by all `logging` resources, except if the `logging` defines a
+`watchNamespaces` field. In that case _flow_ and _output_ resources are only combined into a logging configuration if 
+they are all in the defined list of watched namespaces.
+
+_Clusterflow_ and _clusteroutput_ resources (which are actually namespaced resources with multi-namespace capabilities) are by default
+processed only in the logging control namespace, which is the namespace where the fluentd/syslogng statefulset and fluentbit daemonset are created.
+Additionally there is an option to process these from all namespaces.
+
+### Limitations
+
+1. The collector (fluentbit) collects all logs and sends them to the aggregator (fluentd/syslog-ng). There is no option to filter or route log records based on any criteria there. 
+
+    In case someone defines multiple tenants, they have to either 
+    - set up a routing logging that collects and forwards all logs to separate standalone logging aggregators 
+      (a logging resource that has no fluentbit collector defined). 
```

Comment:
```suggestion
    In case someone defines multiple tenants, they have to either
    - set up a routing logging that collects and forwards all logs to separate standalone logging aggregators
      (a logging resource that has no fluentbit collector defined).
```